components/project-list/project-list.jsx

```diff
@@ -16,28 +16,48 @@ export default React.createClass({
       displayBatchIndex: 1
     };
   },
+  getDefaultProps() {
+    return {
+      params: {
+        featured: null,
+        issue: null,
+        search: null,
+        ids: null
+      }
+    };
+  },
   componentDidMount() {
     this.fetchData(this.props.params);
   },
   componentWillReceiveProps(nextProps) {
+    // Reset state
+    this.resetState();
```

Comment:
Spinners are nice. 

I haven't read this whole thread, but if we get stuck on this, I suggest we consider adding a _Go!_ button. Find as you type is only more awesome if it feels more awesome.